app/models/compute_resources/foreman/model/ovirt.rb

```diff
@@ -288,6 +290,18 @@ def create_vm(args = {})
       vm
     end
 
+    def id_name_substitution(argument_list, argument)
+      argument_list.detect { |a| a.name == argument }.try(:id) || argument
+    end
+
+    def get_verified_argument(argument_list, argument)
+      if argument_list.detect { |a| a.name == argument }.nil? && argument_list.detect { |a| a.id == argument }.nil?
+        raise Foreman::Exception.new("#{argument} is not valid, enter id or name")
+      else
+        id_name_substitution(argument_list, argument)
```

Comment:
If this is the only usage I'd avoid creating a one line method for it.

Reply:
I agree I will add the line to this function instead of having another method.